### PR TITLE
feat: gitleaks precommit hook (SDKCF-2442)

### PR DESCRIPTION
**Pull request by apparition47**
# Why?

To prevent devs from committing secrets to their repo.

Includes override of config file (version https://github.com/zricethezav/gitleaks/commit/9326f35380636bcbe61e94b0584d1618c4b5c2c2)

## To install precommit hook
Note: will refuse to install if there are preexisting pre-commit hooks
```bash
brew install gitleaks
fastlane ios setup_project
```

## To remove precommit hook
```bash
rm .git/hooks/pre-commit
rm .git/hooks/.gitleaks.toml
```

### CODE REVIEW

**Review by imaginaris — APPROVED** [review]

**Review by donnie-jp — APPROVED** [review]
Very nice! 👍

**Comment by apparition47** [comment]
Had to include an override of the default `.gitleaks.yaml` config to get around the bogus UUID values used in our tests

File: fastlane/actions/setup_git_hooks.rb [hunk 1]
```diff
@@ -1,24 +1,561 @@
...
+entropy = 3.7
+secretGroup = 4
+
+[rules.allowlist]
+description = "allow UUIDs"
+regexes = [
+    '''[A-F0-9]{8}-[A-F0-9]{4}-4[A-F0-9]{3}-[89AB][A-F0-9]{3}-[A-F0-9]{12}'''
+]
+
+[allowlist]
```
Comment (donnie-jp) [inline fastlane/actions/setup_git_hooks.rb:515]: So to have an allowlist we need to duplicate the entire rules config? That's not so good.

**Review by donnie-jp — APPROVED** [review]

File: fastlane/actions/setup_git_hooks.rb (hunk 1 above)
Comment (apparition47) [reply fastlane/actions/setup_git_hooks.rb:515]: Yep, couldn't find a way to extend the defaults (the [author seems to favour devs use their own ruleset over the defaults](https://github.com/zricethezav/gitleaks/issues/543#issuecomment-831916731))

**Review by imaginaris — APPROVED** [review]

File: fastlane/actions/setup_git_hooks.rb (hunk 1 above)
Comment (apparition47) [reply fastlane/actions/setup_git_hooks.rb:515]: Others are asking for a similar feature--to extend the default config (https://github.com/zricethezav/gitleaks/issues/741). I'll merge this for now then we can change this if gitleaks includes this feature in the future

**Pull request merged**